### PR TITLE
feat(nx-plugin): added support for vitest test runner

### LINKED ISSUE #22882: [Nx Plugin] Vitest support for nx plugins generators
### Current Behavior

Hi folks...

It's well known that, at least for now, nx plugins needs to be CommonJS only, but those plugins can rely on other publishable packages from the same workspace.

Those publishable libraries, needs to be CommonJS too, but for library generators we have the option to use Vite as bundler and Vitest as unit test runner...

Considering this, would be great to have an option to use Vitest as test runner for plugins, in that way we can remove jest dependencies...

### Expected Behavior

Provide also Vitest as test runner in plugin generator

### GitHub Repo

_No response_

### Steps to Reproduce

N/A

### Nx Report

```shell
N/A
```

### Failure Logs

_No response_

### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I created also a discussion for this https://github.com/nrwl/nx/discussions/22834

### PR DESCRIPTION
**Pull request by gperdomor**
## Current Behavior

Nx Plugin do not have the option to use Vitest as test runner

## Expected Behavior

This PR allow us to use Vitest as test runner in plugins projects.

Vitest for e2e targets is still missing and a new PR will be raised after https://github.com/nrwl/nx/issues/29139 is fixed

## Related Issue(s)

Fixes #22882

### CODE REVIEW

**Review by isaacplmann — APPROVED** [review]
Docs changes look good

**Comment by gperdomor** [comment]
Hi @FrozenPandaz, can you please take a look at this when you have some time? 🙏🏻

**Comment by gperdomor** [comment]
Folks let me know if any change it's required before merge this 🙏🏻 ...

**Comment by gperdomor** [comment]
Hi folks... Can you merge this please? have been passed two weeks since was approved 🙏🏻

**Comment by isaacplmann** [comment]
I approved it for the docs changes, but @FrozenPandaz needs to review this to decide if the change makes technical sense and if we want to take on the maintenance of the change, etc.

Sorry if my approval communicated more of a positive indicator than it actually carried. I was just approving to make sure that I wasn't blocking the PR.

**Comment by gperdomor** [comment]
Yeah don't worry @isaacplmann and thank you for your quick response...

Just to clarify, my intention with this change is reduce the dependency on jest using vitest everywhere...
I have a workspace with some shared libraries where I'm using Vitest and some plugins too, the idea is use only one testing library... 😄

**Comment by AgentEnder** [comment]
I believe the generated test files for vitest would not make sense? Vitest requires importing describe, it, etc and those are not imported in the specs. If we want to add this support, we need to make sure that running a generator's tests will work out of the box

**Comment by gperdomor** [comment]
@AgentEnder works without importing describe or other stuff from vitest, I published a local version using verdaccio and create a workspace and new plugins project without issues

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**
